taxi/tests/test_views_cars.py

```diff
@@ -0,0 +1,152 @@
+
+        car = Car.objects.get()
+
+        self.assertEqual(car.model, "testmodel")
+        self.assertEqual(car.manufacturer.name, "test1")
+        self.assertEquals(list(car.drivers.all()), [self.driver])
+
+
+class CarUpdateTest(TestCase):
+
+    def setUp(self) -> None:
+        self.driver = get_user_model().objects.create_user(
+            username="test_driver",
+            first_name="test_fn",
+            last_name="test_ln",
+            password="test_pass123",
+            license_number="TES12345"
+        )
+        self.driver2 = get_user_model().objects.create_user(
+            username="test_driver2",
+            first_name="test_fn2",
+            last_name="test_ln2",
+            password="test_pass12322",
+            license_number="TES1234522"
+        )
+
+        self.client.force_login(self.driver)
+
+        self.manufacturer = Manufacturer.objects.create(
+            name="test1",
+            country="TestCountry1"
+        )
+        self.manufacturer2 = Manufacturer.objects.create(
+            name="test2",
+            country="TestCountry2"
+        )
+
+        self.instance = Car.objects.create(
+            model="testmodel",
+            manufacturer=self.manufacturer
+        )
+        self.instance.drivers.add(self.driver)
+
+    def test_car_update(self):
+        url = reverse("taxi:car-update", kwargs={"pk": self.instance.id})
+        data = {
+            "model": "testmodel2",
+            "manufacturer": self.manufacturer2.id,
+            "drivers": [self.driver2.id]
+        }
+
+        response = self.client.post(url, data)
+
+        self.instance.refresh_from_db()
+
+        self.assertEquals(response.status_code, 302)
+        self.assertEquals(self.instance.model, "testmodel2")
+        self.assertEquals(self.instance.manufacturer, self.manufacturer2)
+        self.assertEquals(list(self.instance.drivers.all()), [self.driver2])
+
+
+class CarDeleteTest(TestCase):
+
+    def setUp(self) -> None:
+        self.driver = get_user_model().objects.create_user(
+            username="test_driver",
+            first_name="test_fn",
+            last_name="test_ln",
+            password="test_pass123",
+            license_number="TES12345"
+        )
+
+        self.client.force_login(self.driver)
+
+        self.manufacturer = Manufacturer.objects.create(
+            name="test1",
+            country="TestCountry1"
+        )
+        self.instance = Car.objects.create(
+            model="testmodel",
+            manufacturer=self.manufacturer
+        )
+        self.instance.drivers.add(self.driver)
+
+    def test_car_delete(self):
+        url = reverse("taxi:car-delete", kwargs={"pk": self.instance.id})
+
+        response = self.client.post(url)
+
+        self.assertEquals(response.status_code, 302)
+        self.assertFalse(Car.objects.filter(id=self.instance.id).exists())
+        self.assertRedirects(response, reverse("taxi:car-list"))
```

Comment:
there is no need to create a new class for every test 
u can create PrivateCarTests class for all of this tests, also it will make ur code more DRY

Reply:
also what about toggle assign to car/search feature ?